lnd_test.go

```diff
@@ -5661,19 +5661,459 @@ func testRevokedCloseRetributionRemoteHodl(net *lntest.NetworkHarness,
 	}
 }
 
+// testDataLossProtection tests that if one of the nodes in a channel
+// relationship lost state, they will detect this during channel sync, and the
+// up-to-date party will force close the channel, giving the outdated party the
+// oppurtunity to sweep its output.
+func testDataLossProtection(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+	const (
+		timeout     = time.Duration(time.Second * 10)
+		chanAmt     = maxBtcFundingAmount
+		paymentAmt  = 10000
+		numInvoices = 6
+		defaultCSV  = uint32(4)
+	)
+
+	// Carol will be the up-to-date party.
+	carol, err := net.NewNode("Carol", []string{"--nolisten"})
+	if err != nil {
+		t.Fatalf("unable to create new carol node: %v", err)
+	}
+	defer func() {
+		if err := net.ShutdownNode(carol); err != nil {
+			t.Fatalf("unable to shutdown carol: %v", err)
+		}
+	}()
+
+	// Dave will be the party losing his state.
+	dave, err := net.NewNode("Dave", []string{})
+	if err != nil {
+		t.Fatalf("unable to create new node: %v", err)
+	}
+	defer func() {
+		if err := net.ShutdownNode(dave); err != nil {
+			t.Fatalf("unable to shutdown dave: %v", err)
+		}
+	}()
+
+	// We must let Dave communicate with Carol before they are able to open
+	// channel, so we connect Dave and Carol,
+	if err := net.ConnectNodes(ctxb, carol, dave); err != nil {
```

Comment:
This'll fail after rebasing to master as `carol` was started in `--nolisten` mode.